shared/web_features_manifest_util.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2024 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import (
+	"context"
+	"io"
+)
+
+// GetWPTWebFeaturesManifest is the entrypoint for handling web features manifest downloads.
+// It includes two generic steps: Downloading and Parsing.
+func GetWPTWebFeaturesManifest(ctx context.Context, downloader WebFeaturesManifestDownloader, parser WebFeatureManifestParser) (WebFeaturesData, error) {
+	manifest, err := downloader.Download(ctx)
+	if err != nil {
+		return nil, nil
```

Comment:
```suggestion
		return nil, err
```

Reply:
Good catch! I added a test to catch this too.